src/Components/Web/src/WebEventData.cs

```diff
@@ -3,57 +3,57 @@
 
 using System;
 using System.Text.Json;
-using System.Threading.Tasks;
 using Microsoft.AspNetCore.Components.Rendering;
-using Microsoft.JSInterop;
 
 namespace Microsoft.AspNetCore.Components.Web
 {
     /// <summary>
-    /// Provides mechanisms for dispatching events to components in a <see cref="Renderer"/>.
+    /// For framework use only.
     /// </summary>
-    public static class RendererRegistryEventDispatcher
+    public class WebEventData
     {
+        // This class represents the second half of parsing incoming event data,
+        // once the type of the eventArgs becomes known.
+
         /// <summary>
         /// For framework use only.
         /// </summary>
-        [JSInvokable(nameof(DispatchEvent))]
-        public static Task DispatchEvent(BrowserEventDescriptor eventDescriptor, string eventArgsJson)
+        public WebEventData(string eventDescriptorJson, string eventArgsJson)
+            : this(Deserialize<WebEventDescriptor>(eventDescriptorJson), eventArgsJson)
```

Comment:
Could we share the the parsing code in the same fashion we do other bits? That way we don't have to have a public way of deserializing the descriptors and this class can simply represent the data.

Reply:
Thinking about this more, I think we should make `WebEventData` properly `public` and not hide behind pubternal. It will be needed for updating ComponentsElectron. Since it's only a special way of parsing JSON, it doesn't give the user any extra special powers.

Reply:
As per other comments, we've agreed to make it internal and shared source